addons/account/tests/test_account_payment.py

```diff
@@ -512,3 +513,38 @@ def test_payments_copy_data(self):
 
         self.assertEqual(duplicate_payment_1.amount, payment_1.amount)
         self.assertEqual(duplicate_payment_2.amount, payment_2.amount)
+
+    def test_payments_invoice_payment_state_without_outstanding_accounts(self):
+        """ Ensures that, without outstanding accounts set on the bank journal payment method,
+            the payment of the invoice still gets a journal entry in community edition """
+        def register_payment(move, amount, is_community):
+            def patched_get_invoice_in_payment_state(self):
+                return 'paid' if is_community else 'in_payment'
+
+            with patch.object(type(self.env['account.move']), '_get_invoice_in_payment_state', patched_get_invoice_in_payment_state):
+                return self.env['account.payment.register'].with_context(
+                    active_model='account.move',
+                    active_ids=move.ids
+                ).create({'amount': amount})._create_payments()
+
+        # Remove the outstanding account on the payment method line to avoid generating a journal entry on the payment
+        self.company_data['default_journal_bank'].inbound_payment_method_line_ids.payment_account_id = self.env['account.account']
+
+        invoice_1 = self.env['account.move'].create([{
+            'move_type': 'out_invoice',
+            'partner_id': self.partner_a.id,
+            'date': '2024-01-01',
+            'invoice_line_ids': [Command.create({
+                'name': 'test',
+                'quantity': 1,
+                'price_unit': 100.0,
+            })],
+        }])
+        invoice_1.action_post()
+        register_payment(invoice_1, 100.0, is_community=True)
+        self.assertTrue(invoice_1.matched_payment_ids.move_id)
```

Comment:
Isn't this the whole purpose of this commit though?

Reply:
But there should be an oustanding in community :thinking: 

Isn't this the whole purpose of this commit?

Reply:
Yes there is indeed an outstanding, but it can't therefore get the `paid` status directly (without manual validation I guess). The payment can only get this status if no outstanding is set on the payment, but since I add one, it changes the automatic switch to `paid` in community. I've removed this condition but can't really tell if it will have other impacts